controllers/databaseengine_controller.go

```diff
@@ -70,89 +83,201 @@ func (r *DatabaseEngineReconciler) Reconcile(ctx context.Context, req ctrl.Reque
 			Namespace: req.NamespacedName.Namespace,
 		},
 	}
-	_, err := controllerutil.CreateOrUpdate(ctx, r.Client, dbEngine, func() error {
+	if _, err := controllerutil.CreateOrUpdate(ctx, r.Client, dbEngine, func() error {
 		dbEngine.Spec.Type = engineType
 		return nil
-	})
-	if err != nil {
+	}); err != nil {
 		return ctrl.Result{}, err
 	}
 
+	requeue := false
+	if done, err := r.handleOperatorUpgrade(ctx, dbEngine); err != nil {
+		if !errors.Is(err, errInstallPlanNotFound) {
+			return ctrl.Result{}, err
+		}
+		// We could not find the InstallPlan for the operator upgrade,
+		// so we will fallthrough since we'd still want the engine status to be updated.
+		// We will still requeue to check for the InstallPlan later.
+		requeue = true
+		logger.Error(err, "Failed to handle operator upgrade")
```

Comment:
I think we could make this message be more specific and say we don't have the IP yet.

Reply:
Sounds good, updated..